internal/svc/admin_svc.go

```diff
@@ -233,6 +235,35 @@ func (cas *CerbosAdminService) DeleteSchema(ctx context.Context, req *requestv1.
 	return &responsev1.DeleteSchemaResponse{}, nil
 }
 
+func (cas *CerbosAdminService) ReloadStore(ctx context.Context, _ *requestv1.ReloadStoreRequest) (*responsev1.ReloadStoreResponse, error) {
+	log := ctxzap.Extract(ctx)
+	if err := cas.checkCredentials(ctx); err != nil {
+		return nil, err
+	}
+
+	_, err, shared := sfGroup.Do("reload", func() (interface{}, error) {
+		rs, ok := cas.store.(storage.ReloadableStore)
+		if !ok {
+			return nil, status.Error(codes.Unimplemented, "Configured store is not reloadable")
+		}
```

Comment:
I think we can move this check out of the singleflight group. It's a cheaper to do this check before instantiating the machinery needed for singleflight.